### PR TITLE
Colony Guns Part 1: Sidearms

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

The Beretta has been buffed significantly, and now resembles a higher-capacity M4A3 with less bullet damage. It also has a small chance to spawn as the 93R police issue burstfire variant.

The Hipower has been completely reworked and redesigned into a new weapon, with new sprites, new inhands, new lore, new stats. It fires 14 bullets that push back (not knock back) close targets when struck.

The Deagle's normal shots (unobtainable by COs) throw back any hit targets like heavy m44, and deal great damage. However, the gun is very inaccurate.

The PK9 has recieved an accuracy buff as it was hitting nothing.

The KT-42 has been made into a cross between a mod88 and a VP, quickly firing strong, inaccurate AP bullets.

The unfinished RXM5-EVA pistol has been finished! With its own unique codersprite for inhands. It fires weak non-AP red laser bolts that deal some stamina damage and have a small chance of weakly igniting the target.

Added unused pistol belts for the new highpower.

To make room for AMMO_LASER, AMMO_XENO_(TOX|ACID|BONE) have been consolidated into AMMO_ACIDIC and AMMO_XENO.

Laser armor is now actually used, against the eva pistol and the laser uzi.

Added attachment sprites for every bayonet type.

The CMB revolver has received a complete overhaul. It has completely refreshed alien isolation sounds, has gotten a complete stat pass, and now fires by default hollowpoint rounds (Like in Isolation!)

The S&W 'model 48' revolver is very powerful, but also inaccurate. You will be able to focus on your shots better if you do some cool tricks with it first?

Nagant: Shrapnel shot has unique handful sprites.

BR blasters have USCM pistol placeholder inhands.

## Why It's Good For The Game

Colony guns are so useless even mappers or 'survivor devs' don't bother with them and just add flamers and tac shotties everywhere.

## Changelog

:cl:
add:The Beretta has been buffed significantly, and now resembles a higher-capacity M4A3 with less bullet damage. It also has a small chance to spawn as the 93R police issue burstfire variant.
add:The Hipower has been completely reworked and redesigned into a new weapon, with new sprites, new inhands, new lore, new stats. It fires 14 bullets that push back (not knock back) close targets when struck.
add:The Deagle's normal shots (unobtainable by COs) throw back any hit targets like heavy m44, and deal great damage. However, the gun is very inaccurate.
add:The PK9 has recieved an accuracy buff as it was hitting nothing.
add:The KT-42 has been made into a cross between a mod88 and a VP, quickly firing strong, inaccurate AP bullets.
add:The unfinished RXM5-EVA pistol has been finished! With its own unique codersprite for inhands. It fires weak non-AP red laser bolts that deal some stamina damage and have a small chance of weakly igniting the target. Only really accessible in New Varadero for now.
imageadd: Added attachment sprites for every bayonet type.
balance:The CMB revolver has received a complete overhaul. It has completely refreshed alien isolation sounds, has gotten a complete stat pass, and now fires by default hollowpoint rounds (Like in Isolation!)
add:The S&W 'model 48' revolver is very powerful, but also inaccurate. You will be able to focus on your shots better if you do some cool tricks with it first?
add:Nagant: Shrapnel shot has unique handful sprites.
add:BR blasters have USCM pistol placeholder inhands.
/:cl:

### CODE REVIEW

**Comment by MusselMain** [comment]
Awesome concept, too bad it does nothing for the CMB revolver

**Comment by carlarctg** [comment]
> Awesome concept, too bad it does nothing for the CMB revolver

Note the part 1, revolvers are likely part 2.

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -2484,7 +2530,7 @@
 	name = "neurotoxic air splash"
 	effect_power = XENO_NEURO_TIER_1
 	spit_cost = 50
-	flags_ammo_behavior = AMMO_XENO_TOX|AMMO_IGNORE_RESIST
+	flags_ammo_behavior = AMMO_XENO|AMMO_IGNORE_RESIST
```
Comment (harryob) [inline code/modules/projectiles/ammo_datums.dm:2533]: is commenting out truly how we must deal with dead code

Comment (harryob) [reply code/modules/projectiles/ammo_datums.dm:2533]: i'm not saying u have to change this btw it's just lame asf

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -166,9 +166,11 @@ Defined in conflicts.dm of the #defines folder.
 		// Apply bullet traits from attachment to gun's current projectile
 		G.in_chamber.apply_bullet_trait(L)
 
-/obj/item/attachable/proc/Detach(var/obj/item/weapon/gun/G)
+/obj/item/attachable/proc/Detach(var/mob/user, var/obj/item/weapon/gun/G)
```
Comment (harryob) [inline code/modules/projectiles/gun_attachables.dm:169]: G -> detach_from or w/e u should've known this was gonna happen literally redefining the args......

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -121,7 +121,7 @@ Defined in conflicts.dm of the #defines folder.
 	*/
 	if(G.attachments[slot])
 		var/obj/item/attachable/A = G.attachments[slot]
-		A.Detach(G)
+		A.Detach(null, G)
```
Comment (harryob) [inline code/modules/projectiles/gun_attachables.dm:124]: G should be a named arg without the null... we hate null args carlarc

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -742,7 +748,7 @@ Defined in conflicts.dm of the #defines folder.
 	. = ..()
 	G.AddElement(/datum/element/drop_retrieval/gun, retrieval_slot)
 
-/obj/item/attachable/magnetic_harness/Detach(var/obj/item/weapon/gun/G)
+/obj/item/attachable/magnetic_harness/Detach(var/mob/user, var/obj/item/weapon/gun/G)
```
Comment (harryob) [inline code/modules/projectiles/gun_attachables.dm:751]: ur gonna hate me

File: code/modules/projectiles/gun_attachables.dm [hunk 5]
```diff
@@ -765,7 +771,7 @@ Defined in conflicts.dm of the #defines folder.
 	G.attachable_offset["under_y"] = 12
 
 
-/obj/item/attachable/magnetic_harness/lever_sling/Detach(var/obj/item/weapon/gun/G)
+/obj/item/attachable/magnetic_harness/lever_sling/Detach(var/mob/user, var/obj/item/weapon/gun/G)
```
Comment (harryob) [inline code/modules/projectiles/gun_attachables.dm:774]: :3

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_attachables.dm (hunk 2 above)
Comment (carlarctg) [reply code/modules/projectiles/gun_attachables.dm:169]: erm i hate oyu

File: code/modules/projectiles/gun_attachables.dm (hunk 3 above)
Comment (carlarctg) [reply code/modules/projectiles/gun_attachables.dm:124]: ****just like i hate you

File: code/modules/projectiles/gun_attachables.dm (hunk 4 above)
Comment (carlarctg) [reply code/modules/projectiles/gun_attachables.dm:751]: RAAAAAAAAAAAAAAAAAAAGH

**Review by harryob — APPROVED** [review]
code

File: code/modules/projectiles/guns/revolvers.dm [hunk 6]
```diff
@@ -479,32 +482,39 @@
...
-	accuracy_mult = BASE_ACCURACY_MULT
-	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_3
-	scatter = SCATTER_AMOUNT_TIER_6
-	scatter_unwielded = SCATTER_AMOUNT_TIER_6
-	damage_mult = BASE_BULLET_DAMAGE_MULT
+	fire_delay = FIRE_DELAY_TIER_6
+	accuracy_mult = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_7
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_7
+	scatter = SCATTER_AMOUNT_TIER_5
+	damage_mult = BASE_BULLET_DAMAGE_MULT * 2
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/revolvers.dm:506]: too much

File: code/modules/projectiles/guns/revolvers.dm [hunk 7]
```diff
@@ -479,32 +482,39 @@
...
 	recoil = 0
 	recoil_unwielded = 0
 
 /obj/item/weapon/gun/revolver/small/unique_action(mob/user)
-	revolver_trick(user)
+	var/result = revolver_trick(user)
+	if(result)
+		to_chat(user, SPAN_NOTICE("Your badass trick inspires you. Your next few shots will be focused!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/revolvers.dm:513]: I do not wish for the otherwise soulful tricks to be turned into a powergame tool

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by kugamo — CHANGES_REQUESTED** [review]
https://discord.com/channels/150315577943130112/746325423289401395/1053159057403433071

I put my sprite review here, minor stuff.. But when fixed should be good.

**Review by kugamo — APPROVED** [review]
SPRITES APPROVED

**Pull request merged**